pkg/agent/supportbundlecollection/support_bundle_controller.go

```diff
@@ -0,0 +1,439 @@
+	klog.Info("Starting watch for SupportBundleCollections")
+	antreaClient, err := c.antreaClientGetter.GetAntreaClient()
+	if err != nil {
+		klog.ErrorS(err, "Failed to get antrea client")
+		return
+	}
+	options := metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("nodeName", c.nodeName).String(),
+	}
+	watcher, err := antreaClient.ControlplaneV1beta2().SupportBundleCollections().Watch(context.TODO(), options)
+	if err != nil {
+		klog.ErrorS(err, "Failed to start watch for SupportBundleCollections")
+		return
+	}
+	// Watch method doesn't return error but "emptyWatch" in case of some partial data errors,
+	// e.g. timeout error. Make sure that watcher is not empty and log warning otherwise.
+	if reflect.TypeOf(watcher) == reflect.TypeOf(emptyWatch) {
+		klog.Warning("Failed to start watch for SupportBundleCollections, please ensure antrea service is reachable for the agent")
+		return
+	}
+
+	klog.Info("Started watch for SupportBundleCollections")
+	eventCount := 0
+	defer func() {
+		klog.InfoS("Stopped watch for SupportBundleCollections", "total items received", eventCount)
+		watcher.Stop()
+	}()
+
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return
+			}
+			switch event.Type {
+			case watch.Bookmark:
+				klog.V(2).Info("Received Bookmark event")
+			case watch.Added:
+				c.addSupportBundleCollection(event.Object.(*cpv1b2.SupportBundleCollection))
+				klog.InfoS("Added SupportBundleCollection", "name", event.Object.(*cpv1b2.SupportBundleCollection).Name)
+			case watch.Deleted:
+				c.deleteSupportBundleCollection(event.Object.(*cpv1b2.SupportBundleCollection))
+				klog.InfoS("Deleted SupportBundleCollection", "name", event.Object.(*cpv1b2.SupportBundleCollection).Name)
+			default:
+				klog.Errorf("Unknown event: %v", event)
+				return
+			}
+			eventCount++
+		}
+
+	}
+}
+
+func (c *SupportBundleController) addSupportBundleCollection(supportbundle *cpv1b2.SupportBundleCollection) {
+	c.supportBundleCollectionsMutex.Lock()
+	defer c.supportBundleCollectionsMutex.Unlock()
+
+	c.supportBundleCollections[supportbundle.Name] = supportbundle
+	c.queue.Add(supportbundle.Name)
+}
+
+func (c *SupportBundleController) deleteSupportBundleCollection(supportbundle *cpv1b2.SupportBundleCollection) {
+	c.supportBundleCollectionsMutex.Lock()
+	defer c.supportBundleCollectionsMutex.Unlock()
+
+	delete(c.supportBundleCollections, supportbundle.Name)
+	c.queue.Add(supportbundle.Name)
+}
+
+func (c *SupportBundleController) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.InfoS("Starting", "controllerName", controllerName)
+	defer klog.InfoS("Shutting down", "controllerName", controllerName)
+
+	go wait.NonSlidingUntil(c.watchSupportBundleCollections, 5*time.Second, stopCh)
+
+	go wait.Until(c.worker, time.Second, stopCh)
+	<-stopCh
+}
+
+func (c *SupportBundleController) GetSupportBundleCollectionCondition() v1alpha1.SupportBundleCollectionConditionType {
```

Comment:
No need to make this function public since it is only accessed inside the same package

Reply:
I have removed this function since it is only used for unit test.